### PR TITLE
(plugin) hardware::devices::hms::netbiter::argos::restapi - new

### PR DESCRIPTION
**Pull request by thibaults-centreon**
./centreon-plugins/src/centreon_plugins.pl --plugin=hardware::devices::hms::netbiter::argos::restapi::plugin --mode=sensors --access-key='xxxx' --system-id='1234567890AB' --filter-device='ZONE1' --verbose 
OK: All sensors are OK | '01 - temp set point#sensor.numeric.value'=-22.0°C;;;; '02 - current temperature#sensor.numeric.value'=-20.2°C;;;; '04 - current battery temperature#sensor.numeric.value'=-31.0°C;;;;
Device: 10 - ZONE1, Name: 01 - temp set point, Value: -22.0°C (2023-04-14 15:00:00 CEST)
Device: 10 - ZONE1, Name: 02 - current temperature, Value: -20.2°C (2023-04-14 15:15:00 CEST)
Device: 10 - ZONE1, Name: 04 - current battery temperature, Value: -31.0°C (2023-04-14 15:15:00 CEST)

### CODE REVIEW

File: src/hardware/devices/netbiter/argos/restapi/plugin.pm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+
+    $self->{version} = '1.0';
+    $self->{modes} = {
+        'alarms'       => 'hardware::devices::netbiter::argos::restapi::mode::alarms',
+        'discovery'    => 'hardware::devices::netbiter::argos::restapi::mode::discovery',
+        'list-sensors' => 'hardware::devices::netbiter::argos::restapi::mode::listsensors',
+        'sensor-value' => 'hardware::devices::netbiter::argos::restapi::mode::sensorvalue'
```
Comment (garnier-quentin) [inline src/hardware/devices/netbiter/argos/restapi/plugin.pm:37]: you could use the name: ```sensors```

Comment (thibaults-centreon) [reply src/hardware/devices/netbiter/argos/restapi/plugin.pm:37]: you're right, forgot to rename it after making it "multi-instances compatible"

File: src/hardware/devices/netbiter/argos/restapi/mode/sensors.pm [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'sensors', type => 1, message_multiple => 'All sensors are OK' },
+    ];
+
+    $self->{maps_counters}->{sensors} = [
+        { label => 'sensor-value', nlabel => 'sensor.numeric.value', set => {
```
Comment (garnier-quentin) [inline src/hardware/devices/netbiter/argos/restapi/mode/sensors.pm:49]: For the nlabel, it should be: ```sensor.reading.count``` (to fit with all plugins).

File: packaging/centreon-plugin-Hardware-Devices-Netbiter-Argos-Restapi/rpm.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "dependencies": [
```
Comment (garnier-quentin) [inline packaging/centreon-plugin-Hardware-Devices-Netbiter-Argos-Restapi/rpm.json:2]: missing DateTime

**Review by lucie-dubrunfaut — APPROVED** [review]

**Pull request merged**